server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainIT.java

```diff
@@ -1150,6 +1152,42 @@ public void testCannotAllocateStaleReplicaExplanation() throws Exception {
         }
     }
 
+    public void testExplainRolesOutput() throws Exception {
+        logger.info("--> starting 1 node with 'roles' setting specified");
+        Settings roleSettings = Settings.builder().putList("node.roles", Arrays.asList("master", "data_hot", "ingest")).build();
```

Comment:
NIT: This is not important here, but please generally prefer `List.of(..)` to `Arrays.asList(..)` as first one creates immutable list while the later allows some modifications